src/infrastructure/network/undiciRequest.ts

```diff
@@ -0,0 +1,26 @@
+import { Client } from 'undici';
+import { RequestT } from './types';
+
+const request: RequestT = async ({ basePath, authorization, method, path, body }) => {
+  const client = new Client(basePath);
+  const { body: responseBody } = await client.request({
+    bodyTimeout: 0,
+    headers: {
+      authorization: `Bearer ${authorization}`,
+      'content-type': 'application/json',
```

Comment:
if only json requests (and responses) are supported, we can export methods with the explicit behaviour in the name. (ie: jsonRequest, xmlRequest (only if we need it), {generic}Request)